test/run/idl.mo

```diff
@@ -1,34 +1,37 @@
 //MOC-ENV MOC_UNLOCK_PRIM=yesplease
+//MOC-FLAG -v
 import Prim "mo:⛔";
 
-func serUnit() : Blob = (prim "serialize" : () -> Blob) ();
-func deserUnit(x : Blob) : () = (prim "deserialize" : Blob -> ()) x;
+func serUnit() : Blob = to_candid ();
+func deserUnit(x : Blob) : ?() = from_candid x;
 
-func serNats(x: Nat, y: Nat, z: Nat) : Blob = (prim "serialize" : (Nat,Nat,Nat) -> Blob) (x,y,z);
-func deserNats(x: Blob) : (Nat, Nat, Nat) = (prim "deserialize" : Blob -> (Nat,Nat,Nat)) x;
+func serNats(x: Nat, y: Nat, z: Nat) : Blob = to_candid (x,y,z);
+func deserNats(x: Blob) : ?(Nat, Nat, Nat) = from_candid x;
 
-func serText(x: Text) : Blob = (prim "serialize" : Text -> Blob) x;
-func deserText(x: Blob) : Text = (prim "deserialize" : Blob -> Text) x;
-
-func serBool(x: Bool) : Blob = (prim "serialize" : Bool -> Blob) x;
-func deserBool(x: Blob) : Bool = (prim "deserialize" : Blob -> Bool) x;
+func serBool(x: Bool) : Blob = to_candid (x);
+func deserBool(x: Blob) : ?(Bool) = from_candid x;
 
+func serText(x: Text) : Blob = to_candid (x);
+func deserText(x: Blob) : ?(Text) = from_candid x;
 
 Prim.debugPrint(debug_show (serUnit ()));
 Prim.debugPrint(debug_show (serNats (1,2,3)));
```

Comment:
@crusso This line prints 

`"\44\49\44\4C\00\03\7D\7D\7D\01\02\03"`

So, a message vector of three components, each is a `Nat`.

Reply:
(**Not** a single-component message, with 3-tuple of `Nat`s)